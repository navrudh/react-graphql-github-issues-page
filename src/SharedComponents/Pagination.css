.pagination .page-item a {
  display: block;
  color: black;
  float: left;
  padding: 8px 16px;
  text-decoration: none;
}

.pagination .page-item a:hover {
  background: #eee;
  color: #555;
}

.pagination .page-item.active .page-link, .pagination .page-item a:active {
  background-color: #4CAF50;
  color: white;
}

.sr-only {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0,0,0,0);
  border: 0;
}
