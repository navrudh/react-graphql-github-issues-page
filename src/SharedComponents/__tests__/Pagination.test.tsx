import * as React from 'react';
import * as renderer from 'react-test-renderer';

import Pagination, { PaginationProps } from '../Pagination';
import { mount, shallow } from 'enzyme';

const paginationProps: PaginationProps = {
  onPageChanged: () => {
    return;
  },
  pageLimit: 10,
  pageNeighbours: 3,
  totalRecords: 100
};

describe('Pagination', () => {
  const _component = shallow(<Pagination {...paginationProps} />);

  it('renders and matches our snapshot', () => {
    const component = renderer.create(<Pagination {...paginationProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot renders correctly', () => {
    expect(_component).toMatchSnapshot();
  });

  it('creates pagination buttons', () => {
    expect(_component.find('a')).toHaveLength(9);
  });

  it('displays "Next" button', () => {
    expect(
      _component
        .find('a')
        .map(elem => elem.props()['aria-label'] === 'Next')
        .filter(bool => bool)
    ).toHaveLength(1);
  });

  it('hides "Previous" button', () => {
    expect(
      _component
        .find('a')
        .map(elem => elem.props()['aria-label'] === 'Previous')
        .filter(bool => bool)
    ).toHaveLength(0);
  });

  it('shows "Previous" button', () => {
    _component.find('a').map(
      elem =>
        elem.props()['aria-label'] === 'Next' &&
        elem.simulate('click', {
          // tslint:disable-next-line:no-empty
          preventDefault: () => {}
        })
    );

    expect(
      _component
        .find('a')
        .map(elem => elem.props()['aria-label'] === 'Previous')
        .filter(bool => bool)
    ).toHaveLength(1);
  });

  it('"Next" calls handleMoveRight', () => {
    const component = shallow(<Pagination {...paginationProps} />);

    const instance = component.instance();
    const spy = jest.spyOn(instance, 'handleMoveRight');

    component.find('a').map(
      elem =>
        elem.props()['aria-label'] === 'Next' &&
        elem.simulate('click', {
          // tslint:disable-next-line:no-empty
          preventDefault: () => {}
        })
    );

    component.find('a').map(
      elem =>
        elem.props()['aria-label'] === 'Previous' &&
        elem.simulate('click', {
          // tslint:disable-next-line:no-empty
          preventDefault: () => {}
        })
    );

    component.find('a').map(
      elem =>
        elem.props()['aria-label'] === 'Next' &&
        elem.simulate('click', {
          // tslint:disable-next-line:no-empty
          preventDefault: () => {}
        })
    );

    expect(spy).toHaveBeenCalled();
  });

  it('"Previous" calls handleMoveLeft', () => {
    const component = shallow(<Pagination {...paginationProps} />);

    const instance = component.instance();
    const spy = jest.spyOn(instance, 'handleMoveLeft');

    component.find('a').map(
      elem =>
        elem.props()['aria-label'] === 'Next' &&
        elem.simulate('click', {
          // tslint:disable-next-line:no-empty
          preventDefault: () => {}
        })
    );

    component.find('a').map(
      elem =>
        elem.props()['aria-label'] === 'Previous' &&
        elem.simulate('click', {
          // tslint:disable-next-line:no-empty
          preventDefault: () => {}
        })
    );

    expect(spy).toHaveBeenCalled();
  });
});
