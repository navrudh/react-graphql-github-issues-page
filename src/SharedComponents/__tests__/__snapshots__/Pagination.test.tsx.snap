// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders and matches our snapshot 1`] = `
<div
  className="sc-bwzfXH flex justify-center tc center uTxCW"
>
  <div
    className="bg-dark-blue sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link white"
      href="#"
      onClick={[Function]}
    >
      1
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      2
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      3
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      4
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      5
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      6
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      7
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      aria-label="Next"
      className="link"
      href="#"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        »
      </span>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </div>
  <div
    className="sc-bdVaJa dtc link dim f6 f5-ns pa3 br bl b--light-silver hover-bg-light-blue hover-white bDWFJH"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      10
    </a>
  </div>
</div>
`;

exports[`Pagination snapshot renders correctly 1`] = `
<styled.div>
  <styled.div
    className="bg-dark-blue"
    key="0"
  >
    <a
      className="link white"
      href="#"
      onClick={[Function]}
    >
      1
    </a>
  </styled.div>
  <styled.div
    className=""
    key="1"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      2
    </a>
  </styled.div>
  <styled.div
    className=""
    key="2"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      3
    </a>
  </styled.div>
  <styled.div
    className=""
    key="3"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      4
    </a>
  </styled.div>
  <styled.div
    className=""
    key="4"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      5
    </a>
  </styled.div>
  <styled.div
    className=""
    key="5"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      6
    </a>
  </styled.div>
  <styled.div
    className=""
    key="6"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      7
    </a>
  </styled.div>
  <styled.div
    key="7"
  >
    <a
      aria-label="Next"
      className="link"
      href="#"
      onClick={[Function]}
    >
      <span
        aria-hidden="true"
      >
        »
      </span>
      <span
        className="sr-only"
      >
        Next
      </span>
    </a>
  </styled.div>
  <styled.div
    className=""
    key="8"
  >
    <a
      className="link black"
      href="#"
      onClick={[Function]}
    >
      10
    </a>
  </styled.div>
</styled.div>
`;
